runtime/platform.c

```diff
@@ -114,6 +114,15 @@ void caml_plat_wait(caml_plat_cond* cond)
   check_err("wait", pthread_cond_wait(&cond->cond, cond->mutex));
 }
 
+int caml_plat_timedwait(caml_plat_cond* cond, const struct timespec *deadline)
+{
+  int res;
+  caml_plat_assert_locked(cond->mutex);
+  res = pthread_cond_timedwait(&cond->cond, cond->mutex, deadline);
+  if (res != ETIMEDOUT) check_err("timedwait", res);
```

Comment:
One documentation for `pthread_cond_timedwait` mentions spurious wakeups. In these cases one should probably restart `pthread_cond_timedwait` rather than raise an exception.